### DISCUSSION TITLE
invalid SCRAM verifier error when using Aurora PostgreSQL 13.4 with EdgeDB 1.0-rc4

### DISCUSSION THREAD

**Discussion by jonnyyu** [geldata/gel#3361] [Q&A]
Hi, 

I am trying EdgeDB 1.0-rc4 with Aurora PostgreSQL 13.4 as backend. The edgedb-server seems starts correctly, however I got error when I tried to create instance link.

```
edgedb error: ClientError: AuthenticationError: invalid SCRAM verifier for user 'edgedb'
```
I found [AWS Doc](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.Security.html) mentions SCRAM is not supported in Aurora PostgreSQL, not sure if this is the cause. Is there anyway to workaround this, e.g. can we config edgedb to use `md5` with Aurora backend?

> The Salted Challenge Response Authentication Mechanism (SCRAM) is not supported with Aurora PostgreSQL.
 
Here're the commands I tried:

```
docker run --name edgedb -it --rm  \
    -e EDGEDB_SERVER_BACKEND_DSN=postgres://postgres:<passwd>@<postgres_endpoint> \
    -e EDGEDB_SERVER_DATABASE=edgedb \
    -e EDGEDB_SERVER_USER=edgedb \
    -e EDGEDB_SERVER_PASSWORD=secret \
    -e EDGEDB_SERVER_TLS_CERT_MODE=generate_self_signed \
    -p 5656:5656 \
    edgedb/edgedb:1-rc4

edgedb -P 5656 instance link --trust-tls-cert edgedb_docker
```

Thank you!

**Comment by elprans** [comment]
Hm. This looks more like an authentication issue against EdgeDB, not Postgres.  EdgeDB should correctly negotiate the authentication parameters with Postgres.  I'll see if I can reproduce this.

**Reply by fantix** [reply]
> This looks more like an authentication issue against EdgeDB, not Postgres.

I think that is correct.

I just tried on Aurora PostgreSQL 13.4 and EdgeDB Docker 1.0-rc4, looks like it's working:

```console
(system) fantix@fantix-pro:~$ edgedb -P 5656 instance link --trust-tls-cert edgedb_docker
Specify the host of the server [default: localhost]:
> localhost
Specify the database user [default: edgedb]:
> edgedb
Specify the database name [default: edgedb]:
> edgedb
Trusting unknown server certificate: SHA1:63e9fc9b48931a0009ff38b845714e276d2452fb
Password for 'edgedb': secret
Successfully linked to remote instance. To connect run:
  edgedb -I edgedb_docker
(system) fantix@fantix-pro:~$ edgedb -I edgedb_docker
EdgeDB 1.0-rc.4+c5bc0e8 (repl 1.0.0-rc.6+4b9d831)
Applied /Users/fantix/Library/Application Support/edgedb/cli.toml configuration file
Type \help for help, \quit to quit.
edgedb>
```

However it doesn't seem like you put in the wrong user (`edgedb`) or password (`secret`), because that would raise an `edgedb error: ClientError: AuthenticationError: authentication failed`, instead of an `invalid SCRAM verifier` error.

If you connect to your PostgreSQL backend using a psql REPL, you should be able to see a `password_hash` starting with `SCRAM-SHA-256` through the command `\du+`, this is used to authenticate the EdgeDB user. If this value is damaged, you'll get an `invalid SCRAM verifier` error in the CLI. Usually you don't have to worry about this because EdgeDB manages this for you, but in this case it's worth to find out why.
